patch/cisco-npu-disable-other-bars.patch

```diff
@@ -0,0 +1,90 @@
+From 2bc0b2e84866a7733fc05a34bd8a93e165db99cc Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Tue, 9 Mar 2021 13:54:28 -0800
+Subject: [PATCH] NPU disable other bars
+
+    This is a port of IOS XR patch
+
+        Spitfire-pacific-disable-all-bars-other-than-0.patch
+
+    to Debian kernel. Modifications from IOS XR:
+
+    a. Only issue an INFO message for changed resources
+    b. Include the previous resource data in the INFO message
+    c. Prepare for the possibility of applying patch to
+       future NPU revisions.
+
+    Original patch message follows:
+
+    For Pacific, only bar 0 is valid. Hence disable all other bars as a work
+    around for spitfire
+
+    CDETS: CSCvi15029
+
+    Signed-off-by: Gaurav Gupta <gauragup@cisco.com>
+    Signed-off-by: Subha Keshavaraj <subkesha@cisco.com>
+---
+ drivers/pci/quirks.c | 50 ++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 50 insertions(+)
+
+diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
+index af2149632..932d219e7 100644
+--- a/drivers/pci/quirks.c
++++ b/drivers/pci/quirks.c
+@@ -5462,3 +5462,53 @@ static void apex_pci_fixup_class(struct pci_dev *pdev)
+ }
+ DECLARE_PCI_FIXUP_CLASS_HEADER(0x1ac1, 0x089a,
+ 			       PCI_CLASS_NOT_DEFINED, 8, apex_pci_fixup_class);
++
++#define PCI_DEVICE_ID_LEABA_PACIFIC    0xabcd
++#define PCI_DEVICE_ID_LEABA_GIBRALTAR  0xa001
++#define PCI_DEVICE_ID_LEABA_GRAPHENE   0xa003
++#define PCI_DEVICE_ID_LEABA_PALLADIUM  0xa004
++#define PCI_DEVICE_ID_LEABA_ARGON      0xa005
++#define PCI_DEVICE_ID_LEABA_KRYPTON    0xa006
++
++/*
++ * For Pacific A0, only BAR 0 is valid
++ */
++static void silicon_one_fixup(struct pci_dev *dev)
++{
++	int i;
```

Comment:
This does not follow the Linux coding style. Please check patches with `checkpatch.pl` script in the Linux kernel.

Reply:
Sorry, Firefox display issues in the browser.